src/ut/realmmanager_test.cpp

```diff
@@ -199,6 +986,14 @@ TEST_F(RealmmanagerTest, ManageConnections)
 
   // The connection to peer1 fails. Set the connected flag on the
   // remaining peers. This should just be peer2.
+  // We don't want to test the alarm behaviour here.
+  EXPECT_CALL(*_mock_alarm, get_alarm_state())
+    .WillRepeatedly(Return(AlarmState::UNKNOWN));
+  EXPECT_CALL(*_mock_alarm, set())
+    .Times(testing::AtLeast(0));
+  EXPECT_CALL(*_mock_alarm, clear())
+    .Times(testing::AtLeast(0));
```

Comment:
@eleanor-merry I have added these unrestricted EXPECT_CALL statements as we don't want to test the alarm specific behaviour in the manage_connections UT. Another option is to remove them altogether and have gmock warnings for uninteresting calls. Which one do you prefer?

Reply:
It's better to have the unrestricted expect calls. I'd like the get_alarm_state to return either set or cleared though (as the production code would never return unknown)